src/main/java/org/jenkinsci/account/BoardElection.java

```diff
@@ -30,64 +25,272 @@
     private final Date close;
     private final Application app;
     private final String[] candidates;
-    private final String log;
+    private final String electionLogDir;
+    private final String resultsLogDir;
+    private final String seats;
+    private ArrayList<HashMap<String,String>> result;
+    private ArrayList<String> rawResult;
+    private final int seniority;
+    private int totalVotes;
 
     public BoardElection(Application application, Parameters params) throws Exception {
-        this.app = application;
+        app = application;
+        seniority = params.seniority();
+        result = new ArrayList<HashMap<String, String>>();
         final SimpleDateFormat format = new SimpleDateFormat("yyyy/MM/dd");
         open = format.parse(params.electionOpen());
         close = format.parse(params.electionClose());
         candidates = params.electionCandidates().split(",");
-        log = params.electionLogDir() + "/" + params.electionClose().replaceAll("/","") + ".yaml";
-        File f = new File(log);
-        f.createNewFile();
+        electionLogDir = params.electionLogDir();
+        seats = params.seats();
+        rawResult = new ArrayList<String>();
+
+        LocalDate date = close.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        resultsLogDir = electionLogDir + "/" + date.format(DateTimeFormatter.BASIC_ISO_DATE);
+
+        File dir = new File(resultsLogDir);
+        dir.mkdirs();
     }
 
-    @RequirePOST
-    public HttpResponse doVote(@QueryParameter String vote) throws NamingException, IOException {
+    private String messageDigest(String s) throws NoSuchAlgorithmException {
+        MessageDigest md = MessageDigest.getInstance("MD5");
+        md.update(s.getBytes());
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest);
+    }
+
+    public HttpResponse doResults() throws NamingException, IOException {
+        return HttpResponses.forwardToView(this,"results.jelly");
+    }
 
-        final Myself user = Myself.current();
-        if (user == null) {
-            throw new UserError("Need to be authenticated");
+    public HttpResponse doResult(@QueryParameter String election) throws NamingException, IOException {
+
+        if ( election.isEmpty() || election.equals(".") || election.equals("..")){
+            return new HttpRedirect("/election/results");
+        }
+
+        File directory = new File( electionLogDir + "/" + election );
+        if (!directory.isDirectory() || !directory.exists()){
+            return new HttpRedirect("/election/results");
         }
 
-        if (!isOpen())  {
-            throw new UserError("Election is closed");
+        if (!isResultExist(election) || isOpen(election)){
+            return new HttpRedirect("/election/ongoing");
+        }
+        else{
+            parseRawResult( election );
```

Comment:
I'm a bit concerned about this call as well, and what might invoke `doResult`

Is this invoked on every vote? There's some file ops in `parseRawResult` which concern me here

Reply:
Indeed, parseRawResult() read the file "result.stv" which is the output of the python script. 
And display several informations like the winner,score,...
This function is call every time someone want to display an election result.